interpreter/syntax/ast.ml

```diff
@@ -112,6 +112,7 @@ type loadop = (pack_size * extension) memop
 type storeop = pack_size memop
 
 type simd_loadop = (pack_size * pack_simd) memop
+type simd_loadlaneop = (pack_size * int (* lane index*) ) memop
```

Comment:
I'd drop this one-off type and put the int on the instruction directly.

Reply:
I made it `SimdLaneOp of pack_size memop * int`, is that ok?

Reply:
Another thing to point out, later we will also implement `StoreLane`, which will have the same type as this, so this won't be as once-off anymore.

Reply:
Ah, okay, I wasn't aware of that, thanks! In that case, how about defining a common
```
type simd_laneop = pack_size memop * int
```
(The reason why I think the index shouldn't be inside the memop is that the lane splicing is logically separate from the memory access as such. Though I admit that the distinction is somewhat fuzzy.)

Reply:
Done! 

> (The reason why I think the index shouldn't be inside the memop is that the lane splicing is logically separate from the memory access as such. Though I admit that the distinction is somewhat fuzzy.)

Make sense, load lane is essentially extract then load, so the memory access parameters (address, alignment) doesn't have anything to do with the index at all.